### PR TITLE
JS Reference Type Diagnostic Responses

### PR DESCRIPTION
**Pull request by naributler**

### CODE REVIEW

**Comment by naributler** [comment]
Comfort: 3
Clarity: 3

I used class notes, the solution branch from js-reference-types/lib/lab.js, and Stack Overflow to complete this diagnostic. I referenced this Stack Overflow article: http://stackoverflow.com/questions/5667888/counting-the-occurrences-of-javascript-array-elements

File: lib/diagnostic.js [hunk 1]
```diff
@@ -6,31 +6,101 @@
...
-const normalizeWords = function (/* */) {
-  /* your response here */
+const paragraph =
+  'When in the Course of human events, it becomes necessary for one ' +
+  'people to dissolve the political bands which have connected them ' +
+  'with another, and to assume among the powers of the earth, the ' +
+  'separate and equal station to which the Laws of Nature and of ' +
+  'Nature\'s God entitle them, a decent respect to the opinions of ' +
+  'mankind requires that they should declare the causes which impel ' +
+  'them to the separation.'
```
Comment (benjimelito) [inline lib/diagnostic.js:16]: Inspiring

File: lib/diagnostic.js [hunk 2]
```diff
@@ -6,31 +6,101 @@
...
+const normalizeWords = function (paragraph) {
+  const words = paragraph.split(' ')
+  const normalizedWords = []
+  for (let i = 0; i < words.length; i++) {
+    normalizedWords[i] = words[i].toUpperCase()
+  }
+  return normalizedWords
 }
 
+const reformattedWords = normalizeWords(reformatted)
```
Comment (benjimelito) [inline lib/diagnostic.js:33]: Nice use of functional programming

File: lib/diagnostic.js [hunk 3]
```diff
@@ -6,31 +6,101 @@
...
 // a second argument.
 
-const wordCount = function (/* */) {
-  /* your response here */
+// Solution taken from js-reference-types/lib/lab.js.
+// I'm still trying to understand how line 66 works.
+const wordCount = function wordCount (arr, unique) {
+  let getWords = unique ? uniqueWords : normalizeWords
+  let count = getWords(arr).length
+  return count
```
Comment (benjimelito) [inline lib/diagnostic.js:68]: I think this solution is referencing variables that you do not have in this code, namely `unique`, `normalizeWords`, and `uniqueWords`.

File: lib/diagnostic.js [hunk 4]
```diff
@@ -6,31 +6,101 @@
...
+
 // 4. Write a function that takes a string and returns a dictionary with unique
 // words as keys and a count of each word as the value for that key
-const wordFrequencies = function (/* */) {
-  /* your response here */
+
+// Referenced stack overflow to help with this solution.
+// Need to figure out a simpler way to do this that applies material from class
+const wordFrequencies = function (arr) {
+  const a = []
```
Comment (benjimelito) [inline lib/diagnostic.js:79]: Nice, although this can be done much more simply

**Comment by benjimelito** [comment]
I would be careful about using code that you do not thoroughly understand, but nice work overall. 2/4

**Pull request closed without merging**